### PR TITLE
Added tests for ButtonGroup

### PR DESCRIPTION
**Pull request by JoshBowdenConcepts**
## Summary

Adding unit tests for the ButtonGroup component.

## List of notable changes:

- **added** unit tests for the ButtonGroup component

## What should reviewers focus on?

- Making sure the unit tests cover everything the component does technically

## Steps to test:

1. Open the code and run ```npm run test src/ButtonGroup/ButtonGroup.test.tsx```

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/ButtonGroup/ButtonGroup.test.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const buttonGroupEl = getByTestId('test')
+    expect(buttonGroupEl.tagName).toBe(expectedTag.toUpperCase())
+    expect(buttonGroupEl.classList).toContain(expectedClass)
+  })
+
+  it('renders buttons with the correct element type when buttonAs is set', () => {
+    const expectedTag = 'a'
+
+    const {getByTestId} = render(
+      <ButtonGroup buttonsAs={'a'}>
```
Comment (rezrah) [inline packages/react/src/ButtonGroup/ButtonGroup.test.tsx:32]: Nit for consistency and unnecessary interpolation..

```suggestion
      <ButtonGroup buttonsAs="a">
```

File: packages/react/src/ButtonGroup/ButtonGroup.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  it('renders buttons with the correct size class when buttonSize is set', () => {
+    const expectedClass = 'Button--size-large'
+
+    const {getByTestId} = render(
+      <ButtonGroup buttonSize={'large'}>
+        <Button data-testid="test">Primary Action</Button>
+        <Button>Secondary Action</Button>
+      </ButtonGroup>
+    )
+    const buttonEl = getByTestId('test')
```
Comment (rezrah) [inline packages/react/src/ButtonGroup/ButtonGroup.test.tsx:50]: Could getByRole('button') be more suitable here?

Comment (JoshBowdenConcepts) [reply packages/react/src/ButtonGroup/ButtonGroup.test.tsx:50]: What is the benefit of that over this approach?

File: packages/react/src/ButtonGroup/ButtonGroup.test.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+describe('ButtonGroup', () => {
+  afterEach(cleanup)
+
+  it('renders correctly into the document', () => {
+    const expectedClass = 'ButtonGroup'
+    const expectedTag = 'section'
+
+    const {getByTestId} = render(
+      <ButtonGroup data-testid="test">
```
[the remark this answers is not in the document]
Comment (JoshBowdenConcepts) [reply packages/react/src/ButtonGroup/ButtonGroup.test.tsx]: Fixed in all places I think

File: packages/react/src/ButtonGroup/ButtonGroup.test.tsx (hunk 2 above)
Comment (rezrah) [reply packages/react/src/ButtonGroup/ButtonGroup.test.tsx]: using test ids in testing library is a last resort matcher.. `byRole` is generally preferred approach. If for example Button changed its default tag from `button` to `a`, this test would also then fail. 

Also..

> "getByTestId: The user cannot see (or hear) these, so this is only recommended for cases where you can't match by role or text or it doesn't make sense (e.g. the text is dynamic)."  

https://testing-library.com/docs/queries/about/

Comment (JoshBowdenConcepts) [reply packages/react/src/ButtonGroup/ButtonGroup.test.tsx]: All fixed up this is good to know

**Pull request merged**